src/robotlibcore.py

```diff
@@ -97,7 +98,27 @@ def run_keyword(self, name, args, kwargs=None):
         return self.keywords[name](*args, **(kwargs or {}))
 
     def get_keyword_arguments(self, name):
-        kw = self.keywords[name] if name != '__init__' else self.__init__
+        if robot_version >= '3.2':
+            return self.__new_arg_spec(name)
+        return self.__old_arg_spec(name)
+
+    def __new_arg_spec(self, keyword_name):
+        kw = self.__get_keyword(keyword_name)
+        if kw is None:
+            return [tuple(), ]
```

Comment:
This is invalid return value. The syntax also looks a bit strange when `[()]` could be used instead, but that's not really relevant.

It would probably be better to handle `kw` being none in `get_keyword_arguments` before calling `new/old` methods. I'm not actually sure what would be a good return value in that case. Probably `None` to let Robot handle the fact that we don't know the spec. Returning `[]` would mean that we say the kw accepts no args.

Reply:
I was not sure what should be returned. Other thing that I was considering, that it would be possible to raise an exception. But if `None` is better from Robot Framework side, let's go with that.

Reply:
Items in the list returned from `get_keyword_names` must be strings or tuples with 1-2 items so that the first item is a string.

Isn't it so that `kw` can be `None` only if RF calls `get_keyword_arguments` with some new special value like `__special__`? It's very unlikely to happen in general, but I guess returning `None` as an indication that we don't know is the best solution. This could be tested with unit tests but I'm no certain is it worth the effort.

Reply:
Yes, None is used for other `__special__` values, expect `__init__`.